libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1473,6 +1473,29 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 							return false;
 						}
 
+	// Another special case for `address.code.length`, which should simply call extcodesize
+	if (
+		auto innerExpression = dynamic_cast<MemberAccess const*>(&_memberAccess.expression());
+		member == "length" &&
+		innerExpression &&
+		innerExpression->memberName() == "code" &&
```

Comment:
Can you also check the type of the inner expression? If not, this could create problems when using `using A for B`.

Reply:
Do you mean the assert on line 1485 should be part of the `if`?

Reply:
Is there already a test that checks for the problem that @chriseth stated here?